blog/modules/ROOT/pages/sort-antora-components.adoc

```diff
@@ -0,0 +1,196 @@
+
+[source,yaml]
+----
+name: component-a
+version: 1.0
+title: My Fabulous A
+display-version: Version One Point Zero
+
+asciidoc:
+  attributes:
+    version-sort-key: zanzibar
+----
+
+Since we'll be getting the `version-sort-key` value from the version object rather than the page attributes exposed in the UI model, the key name doesn't need to start with the `page-` prefix.
+
+=== Handlebars template modifications
+
+We're already supplying all the necessary information to the sorting helper, so the `nav-explore.hbs` handlebars template doesn't need significant modification.
+In order to avoid modifying the existing component objects, we'll copy them, so the component comparison code needs to be a little more careful, comparing name rather than objects.
+
+.nav-explore.hbs (with component and version sorting)
+[source,handlebars]
+----
+include::example$sort-components/nav-explore-component-versions.hbs[]
+----
+
+.page-versions.hbs
+[source,handlebars]
+----
+include::example$sort-components/page-versions-component-versions.hbs[]
+----
+
+=== Sorting helper
+
+We need to add version sorting logic to the sorting helper and make it callable separately.
+We'll do this by adding a first parameter describing the type of sort requested.
+
+
+.componentVersionSorter.js
+[source,javascript]
+----
+include::example$sort-components/componentVersionSorter.js[]
+----
+
+The version sorting is in reverse order, just like the default sorting.
+Versions without a `version-sort-order` key are placed last, in their original order.
+
+Component sorting configuration (playbook `asciidoc.attributes.page-component-sort-order` key)::
+  The key value is a comma-separated list of specifiers:
+  Required component:::
+    An explicit component name is placed in the order specified if present and warned if missing.
+  Optional component:::
+    A component name suffixed with '?' is placed in the order specified if present.
+  Hidden component:::
+    A component name prefixed with '!' is hidden if present and warned if missing.
+  Optional hidden component:::
+    A component name prefixed with '!' and suffixed with '?' is hidden if present.
+  Wildcard include:::
+    A '\*' includes all components not otherwise mentioned, in their original order, at the position of the '*'
+  Wildcard hide:::
+    A '!*' causes all components not otherwise mentioned to be hidden.
+Version sorting configuration, per component::
+* Version sorting is specified in the component descriptor with the `asciidoc.attributes.version-sort-key` key.
+* Versions with this key are ordered in reverse lexicographical order of key value.
+* Versions without this key are ordered after those with the key, in the original order.
+* The key value of '!' causes the version to be hidden.
+
+There's a small project demonstrating all this at https://gitlab.com/djencks/antora-component-sorting-example[].
```

Comment:
Is `[]` actually needed? I think it should work without or maybe it's a bug in Asciidoctor 1.5.8 where the regular expression is too greedy and the last dot is part of the URL?

```suggestion
There's a small project demonstrating all this at https://gitlab.com/djencks/antora-component-sorting-example.
```